### PR TITLE
Add tutorial search backend

### PR DESCRIPTION
**Pull request by bilgeyucel**
* Add vercel function to send a POST request dC
* Add example request for `/api/tutorial-search`
* Add env variables to vercel

### CODE REVIEW

File: api/tutorial-search.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        headers: {
+          accept: 'application/json',
+          'content-type': 'application/json',
+          authorization: `Bearer ${process.env.DC_API_KEY}`
+        },
+        body: JSON.stringify({debug: false, queries: [request.query.query]})
+      };
+
+    try {
+      const dcResponse = await fetch(`https://api.cloud.deepset.ai/api/v1/workspaces/${process.env.DC_WORKSPACE_NAME}/pipelines/${process.env.DC_PIPELINE_NAME}/search`, options);
```
Comment (masci) [inline api/tutorial-search.js:13]: nit: you can assign `https://api.cloud.deepset.ai/api/v1/workspaces` to a variable to make this line shorter, something like:
```ts
const dcResponse = await fetch(`${DC_API_URL}/${process.env.DC_WORKSPACE_NAME}/pipelines/${process.env.DC_PIPELINE_NAME}/search`, options);
```

File: themes/haystack/layouts/_default/tutorials.html [hunk 2]
```diff
@@ -254,12 +254,24 @@ <h2>{{ .Params.contribute.title }}</h2>
...
-    const searchTutorials = async() => {
+    const searchTutorials = async (query) => {
       searchModal.classList.remove("default", "not-found")
       searchModalCards.innerHTML = ""
       searchModal.classList.add("loading")
 
-      await sleep(2000)
+      try {
+        const response = await fetch(`/api/tutorial-search?query=${query}`);
+        const result = await response.text();
```
Comment (masci) [inline themes/haystack/layouts/_default/tutorials.html:264]: same as above

**Review by masci — CHANGES_REQUESTED** [review]
Awesome! Left a couple of comments but it looks good!

**Comment by bilgeyucel** [comment]
> Awesome! Left a couple of comments but it looks good!

All comments are resolved! @masci

**Review by masci — APPROVED** [review]
Noice! Let's 🚢

**Pull request merged**